### PR TITLE
update travis.yml and deploy.sh

### PR DESCRIPTION
**Pull request by Madhu-1**
update travis.yaml and deploy.sh to build v0.3.0 images from csi-v0.3 branch instead of master, as we are going to make csi-v1.0 branch as master.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@rootfs @gman0 @j-griffith PTAL

**Comment by humblec** [comment]
@Madhu-1 Thanks ! LGTM

**Comment by Madhu-1** [comment]
@ShyamsundarR PTAL

**Comment by Madhu-1** [comment]
for ref https://github.com/ceph/ceph-csi/issues/266

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ addons:
 language: go
 branches:
   only:
-    - master
+    - csi-v0.3
```
Comment (ShyamsundarR) [inline .travis.yml]: Wouldn't this change prevent the linter and builds in the jobs list below not run against master? IOW, we want these jobs to run, but we do not want to deploy/push the master branch to quay, right? In that case, the change to deploy.sh should be enough, right?

Comment (j-griffith) [reply .travis.yml]: It does seem that we only need the changes to deploy.sh; we do want to keep running linter etc against master.

**Comment by Madhu-1** [comment]
@ShyamsundarR @j-griffith updated the PR, PTAL

File: .travis.yml (hunk 1 above)
Comment (Madhu-1) [reply .travis.yml]: yeah, my bad, we do need to run static checks on the master.

File: .travis.yml [hunk 2]
```diff
@@ -11,8 +11,9 @@ addons:
 language: go
 branches:
   only:
+    - csi-v0.3
     - master
-    - csi-v1.0
+    - csi-v1.0  # remove this once csi-v1.0 becomes master
```
Comment (j-griffith) [inline .travis.yml]: We should chat about this (but not block your PR); what I was getting at before was that ideally IMO there would still be a v1.0 branch as the stable latest release and a master that could be at say "v1.0.1" or "v2.0.0" etc.  So a constantly moving up to date master and however many stable release branches.  I realize that currently that's just v0.3.0 and v1.0.0 and we're early enough in 1.0 that we're not worrying much about updating things here.

**Comment by humblec** [comment]
@j-griffith agree to your thought and we will have stable branches for spec versions. Its just the initial hiccup to get things on track. Thanks.

lgtm.

@gman0 can you merge this PR ?

**Pull request merged**